src/managedwriter/stream_connection.ts

```diff
@@ -86,7 +87,23 @@ export class StreamConnection extends EventEmitter {
     this._connection.on('data', this.handleData);
     this._connection.on('error', this.handleError);
     this._connection.on('close', () => {
-      this.trace('connection closed');
+      this.trace('connection closed', this._lastConnectionError);
+      if (this.hasPendingWrites()) {
+        const retrySettings = this._writeClient._retrySettings;
+        if (
+          retrySettings.enableWriteRetries &&
+          this.isRetryableError(this._lastConnectionError)
+        ) {
+          this.reconnect();
+          this.resendAllPendingWrites();
+        } else {
+          const err = new gax.GoogleError(
+            'aborted due to failed connection, please retry the request'
+          );
+          err.code = gax.Status.ABORTED;
```

Comment:
```suggestion
          err.code = gax.Status.UNAVAILABLE;
```

I think this should be unavailable per [this case chart](https://grpc.github.io/grpc/core/md_doc_statuscodes.html). I really had to think about it though - `ABORTED`, `FAILED_PRECONDITION` and `UNAVAILABLE` are so similar, but I think `ABORTED` is closer to an HTTP 409 whereas `UNAVAILABLE` is more like a 503, which is this use case

Reply:
applied both changes to message and status code. Thanks for the tips and improvements on that